src/sqlfluff/rules/L003.py

```diff
@@ -255,9 +255,12 @@ def _eval(self, segment, raw_stack, memory, **kwargs):
             elif segment.segments or (segment.is_meta and segment.indent_val != 0):
                 # it's not a raw segment or placeholder. Carry on.
                 return LintResult(memory=memory)
+            elif segment.is_type("placeholder"):
```

Comment:
Similar fix as above.

Reply:
Any reason not to merge this into the above code? Especially as it specifically mentions placeholders?

Reply:
I wondered about that. The above comment looks potentially wrong to me. The comment mentions placeholders, but it doesn't seem to do anything placeholder related.

I'm going to have to rework this code before merge, because of some tests it broke. Let's revisit at that time.

I think rather than skip placeholders, we need to process them, but only if there's no regular code on that line.